Sources/CvUnitAI.cpp

```diff
@@ -1866,84 +1866,51 @@ void CvUnitAI::AI_animalMove()
 	PROFILE_FUNC();
 
 	//TB Animal Mod Begin
-#ifdef C2C_BUILD
-	if (!isAnimal())
+	if (canAttack())
 	{
-		FAssertMsg(false, "Subdued animal still has animal AI");
-		OutputDebugString("Choosing animal mission\n");
-		if (AI_heal())
-		{
-			OutputDebugString("Chosen to heal\n");
-			return;
-		}
-
-		if (AI_construct())
+		bool bReckless = GC.getGame().isOption(GAMEOPTION_RECKLESS_ANIMALS);
+		int iAttackProb = GC.getHandicapInfo(GC.getGameINLINE().getHandicapType()).getAnimalAttackProb();
+		int iOddsThreshold = (bReckless ? 0 : 60);
+		//Attack potential targets first. Animals are pretty good at assessing their chances of taking down prey, therefore 60% odds prereq.
+		if ((iOddsThreshold < 1 || iAttackProb > 99 || GC.getGameINLINE().getSorenRandNum(100, "Animal Attack") < iAttackProb)
+			&& AI_attackTargets(2, iOddsThreshold, 0, false, true))
 		{
-			OutputDebugString("Chosen to construct\n");
 			return;
 		}
 
-		if (AI_guardCity())
+		//Then attack elsewhere, recklessness based on animal aggression.
+		int iAggression = 666;
```

Comment:
What is the meaning of this magic number? I note you compare against it below, I'm guessing this is meant to be used as a flag to indicate that iAggression wasn't changed, but it doesn't make the intent clear at all. 
Shouldn't it just be 
`if (bReckless || GC.getGameINLINE().getSorenRandNum(10, "Animal Attack") < iAggression)` 
on line 1891 and 
`if (!bReckless && exposedToDanger(plot(), 60, true) && AI_safety())`
on line 1900?
And if you want to improve it further pull out the iAggression change to a separate function so it would be 
`if (bReckless || GC.getGameINLINE().getSorenRandNum(10, "Animal Attack") < getAggression())`

Reply:
The number indicates that the reckless animal option is active and that the animal should always attack.

Reply:
I guess it could be optimized so that the sorenRandNuimber function isn't called if bReckless is True.

Reply:
Hmm, trying to add the new agression function and get this on compile 
error C2039: 'getMyAggression' : is not a member of 'CvUnitAI'
Do I need to add it to the .h file, or modify something elsewhere?

I added the function right above the AI_animalMove function.
```cpp
int CvUnitAI::getMyAggression()
{
	int iAggression = m_pUnitInfo->getAggression() + getLevel() - 1;
	iAggression *= maxHitPoints() - getDamage();
	iAggression /= 100;
	iAggression *= iAttackProb;
	return iAggression / 100;
}

// Protected Functions...

void CvUnitAI::AI_animalMove()
```

Reply:
lol, I'm missing variables here, forget what I said. lol

Reply:
Ok, figured out how to add the .h part and now it compiles.